src/clr-angular/utils/drag-and-drop/droppable/droppable.spec.ts

```diff
@@ -0,0 +1,480 @@
+            decorateEventWithDropPosition(dragMoveEvent, {pageX: 0, pageY: 0});
+            this.eventBus.broadcast(dragMoveEvent);
+            expect(this.testComponent.dragStartEvent).toEqual(dragStartEvent);
+            expect(this.testComponent.dragMoveEvent).toEqual(dragMoveEvent);
+            expect(this.testComponent.dragEnterEvent).not.toEqual(dragMoveEvent);
+            this.eventBus.broadcast(dragEndEvent);
+            expect(this.testComponent.dropEvent).toBeUndefined();
+        });
+
+        it("should emit on drop if dragEnter has been registered", function() {
+            const dragStartEvent = basicDragEventMock(ClrDragEventType.DRAG_START);
+            const dragMoveEvent = basicDragEventMock(ClrDragEventType.DRAG_MOVE);
+            const dragEndEvent = basicDragEventMock(ClrDragEventType.DRAG_END);
+            this.eventBus.broadcast(dragStartEvent);
+            decorateEventWithDropPosition(dragMoveEvent, {pageX: 500, pageY: 300});
+            this.eventBus.broadcast(dragMoveEvent);
+            expect(this.testComponent.dragStartEvent).toEqual(dragStartEvent);
+            expect(this.testComponent.dragMoveEvent).toEqual(dragMoveEvent);
+            expect(this.testComponent.dragEnterEvent).toEqual(dragMoveEvent);
+            this.eventBus.broadcast(dragEndEvent);
+            expect(this.testComponent.dropEvent).toEqual(dragEndEvent);
+        });
+
+        it("should remove both draggable-match and draggable-over classes on drop", function() {
+            const dragStartEvent = basicDragEventMock(ClrDragEventType.DRAG_START);
+            const dragMoveEvent = basicDragEventMock(ClrDragEventType.DRAG_MOVE);
+            const dragEndEvent = basicDragEventMock(ClrDragEventType.DRAG_END);
+            this.eventBus.broadcast(dragStartEvent);
+            this.fixture.detectChanges();
+            expect(this.droppable.nativeElement.classList.contains("draggable-match")).toBeTruthy();
+            decorateEventWithDropPosition(dragMoveEvent, {pageX: 500, pageY: 300});
+            this.eventBus.broadcast(dragMoveEvent);
+            this.fixture.detectChanges();
+            expect(this.droppable.nativeElement.classList.contains("draggable-over")).toBeTruthy();
+            expect(this.testComponent.dragStartEvent).toEqual(dragStartEvent);
+            expect(this.testComponent.dragMoveEvent).toEqual(dragMoveEvent);
+            expect(this.testComponent.dragEnterEvent).toEqual(dragMoveEvent);
+            this.eventBus.broadcast(dragEndEvent);
+            this.fixture.detectChanges();
+            expect(this.droppable.nativeElement.classList.contains("draggable-match")).toBeFalsy();
+            expect(this.droppable.nativeElement.classList.contains("draggable-over")).toBeFalsy();
+            expect(this.testComponent.dropEvent).toEqual(dragEndEvent);
+        });
+
+        it("should remove both draggable-match and draggable-over classes on drop", function() {
```

Comment:
These unit tests are getting really, really big but I'm not sure what we can do about it.

Anyway, this one and the previous one have the same description and are indeed testing a lot of the same thing, but they have small differences (like testing the `.dropped` class). Can you please update the tests descriptions?

Reply:
Addressed this.